code/game/objects/items/wayfinding.dm

```diff
@@ -155,57 +157,48 @@
 		user.put_in_hands(P)
 		P.owner = user.real_name
 
-/obj/machinery/pinpointer_dispenser/attackby(obj/item/I, mob/user, params)
+/obj/machinery/pinpointer_dispenser/attackby(obj/item/I, mob/living/user, params)
 	if(machine_stat & (BROKEN|NOPOWER))
 		return ..()
 
 	if(istype(I, /obj/item/pinpointer/wayfinding))
 		var/obj/item/pinpointer/wayfinding/WP = I
 
-		to_chat(user, "<span class='notice'>You put \the [WP] in the return slot.</span>")
-
-		var/refundiscredits = FALSE
 		var/itsmypinpointer = TRUE
-
-		//Will they meet the conditions to get a credit reward for recycling?
 		if(WP.owner != user.real_name)
 			itsmypinpointer = FALSE
 
-			if(synth_acc.has_money(refund_amt) && !WP.roundstart) //can it afford to refund and is the pinpointer not from the quirk
-				refundiscredits = TRUE
-				qdel(WP)
-				synth_acc._adjust_money(-refund_amt)
-				var/obj/item/holochip/holochip = new (loc)
-				holochip.credits = refund_amt
-				holochip.name = "[holochip.credits] credit holochip"
-				if(ishuman(user))
-					var/mob/living/carbon/human/customer = user
-					customer.put_in_hands(holochip)
-
-		if(!refundiscredits)
-			qdel(WP)
+		var/is_a_thing = "are [refund_amt] credit\s."
+		if(refund_amt > 0 && synth_acc.has_money(refund_amt) && !WP.roundstart)
+			synth_acc._adjust_money(-refund_amt)
+			var/obj/item/holochip/holochip = new (user.loc)
+			holochip.credits = refund_amt
+			holochip.name = "[holochip.credits] credit holochip"
+			if(ishuman(user))
+				var/mob/living/carbon/human/customer = user
+				customer.put_in_hands(holochip)
```

Comment:
`put_in_hands` is a proc on `/mob`, what's with the typecast?

Reply:
Was it always like that, though? I first coded this a year ago.

Reply:
Yes.

Reply:
I'm not sure if that's true, but I'm not at the computer to check at the moment.

Reply:
So I can't find any evidence this changed looking at the proc's blame but I'm not sure why I'd type it unless it had runtimed first. I remember reading something about a change made for mobs without hands or something. Anyway, I'll remove it.

Reply:
![image](https://user-images.githubusercontent.com/35135081/108036032-8ccad480-6fec-11eb-8648-8d4476f71d09.png)

Reply:
Yes, I already said that.